lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,8 +68,9 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+    Puppet.warning "A reboot is required to progress further. Please reboot the system." if data['rebootrequired'] == true
```

Comment:
We talked about this one. The original one will send a warning, so we decided it would be appropriate here as well. Plus it may give notice that the user needs to do something. As a secondary step I was thinking about how we could remove the warning if the auto notified item was in the catalog.